CMakeLists.txt

```diff
@@ -1,8 +1,19 @@
 cmake_minimum_required(VERSION 3.8.0)
 project(faasm)
 
+# Optional functionality
+option(FAASM_STATIC_LIBS "Statically link Faasm libs" ON)
+option(FAASM_WAMR_SUPPORT "Add support for WAMR" ON)
+option(FAASM_SGX_SUPPORT "Add support for SGX" OFF)
+
+# Switch off Werror only when in SGX mode
+if(FAASM_SGX_SUPPORT)
+    set(CMAKE_CXX_FLAGS "-Wall")
+else()
+    set(CMAKE_CXX_FLAGS "-Wall -Werror")
```

Comment:
It's generally good practice to try and conform to `Werror` if at all possible. Hopefully I can find and fix anything that's breaking it. If it's in the WAMR code then we might just have to switch it off.